p4-16/psa/PSA.mdk

```diff
@@ -1085,30 +1118,30 @@ resubmission mechanism is disabled and no assignments to
 
 ## Packet Recirculation {#sec-recirculate}
 
-Packet recirculation is a mechanism to repeat both ingress and egress
-processing on a packet.
-
-Packet recirculation happens at the end of the egress pipeline. When a
-packet is recirculated, the packet finishes both ingress and egress
-processing and re-enters the ingress parser after being deparsed by
-the egress deparser. The `ingress_port` of the recirculated packet is
-set to `PORT_RECIRCULATE`. The `packet_path` of the recirculated
-packet is changed to `RECIRCULATE`.
+Packet recirculation is a mechanism to repeat ingress processing on a
+packet, after it has completed egress processing.  Unlike a resubmit,
+where the resubmitted packet contents are identical to the packet that
+arrived at the ingress parser, a recirculated packet may have
+different headers than the packet had before recirculation.  This
+could be useful in implementing features such as multiple levels of
+tunnel encapsulation or decapsulation.
+
+Packet recirculation happens at the end of the egress pipeline, but
+whether a packet is recirculated must be chosen during ingress
+processing, by sending the packet to port `PORT_RECIRCULATE`.  When a
```

Comment:
Nit: This sentence is a bit convoluted. I would probably start with the choice being made in ingress and recirculation at the end of egress. However, it is quite clear what's happening even with the current phrasing. Just the 'no but' rule bothering me here a bit :)

Reply:
I've made it two sentences, and reversed the order as you suggested.